### DISCUSSION TITLE
Unable to attach VS Code on remote host's containers and 'Error: Unrecognized Container OS' on remote host.

### DISCUSSION THREAD

**Discussion by nathanielfernandez** [microsoft/vscode-docker#3425] [Q&A]
Hi All,

I'm trying to connect my laptop to a container instance in a remote linux host using vs code. I connect using the remote ssh extension, then after connecting via remote ssh, I attempt to attach to Visual Studio Code. When I do that, I am prompted to upgrade to docker 17.12.0 in windows, however it seems I'm already using the latest version.

<img width="284" alt="image" src="https://user-images.githubusercontent.com/93752762/153977256-f88f8bc0-e682-4ffa-9ef9-0f3c9702ae16.png">

If I select install, the docker desktop gets updated. The docker version on the linux server is Docker version 1.13.1, build 7d71120/1.13.1. While my windows laptop is using Docker Desktop 4.5.0 (74594). Do you know how I could get to 17.12.0?

Also while connected via remote ssh to the remote linux server, I also see there are errors about unrecognized OS when on the docker extension. Is it normal to see this in the docker extension while on a remote server?

<img width="158" alt="image" src="https://user-images.githubusercontent.com/93752762/153977603-27fc2728-b326-4063-9f76-2814d0b15b01.png">

I could at least connect via shell but I don't have an IDE in the remote linux server. I could script my code using 'nano' or other editors while on the remote linux server, but I'd like to use VS code on my windows laptop if possible. If you could share a good approach to achieve it, then thatt would be great. Thanks!

Thanks.

**Comment by bwateratmsft** [comment]
The "Attach VSCode" feature is provided by the Remote - Containers extension. You'd have to file an issue in [their repo](https://github.com/microsoft/vscode-remote-release) for the Docker version issue.

As for the "Unrecognized container OS" issue, this is coming from our file explorer. We use the container OS provided by the inspection result. If you do `docker inspect <containerId>`, what is the output?

**Comment by nathanielfernandez** [comment]
Hi Brandon,

I see. Thank you, I'll file an issue in their repo for the docker version. :) Please see the output of docker inspect:

docker inspect 2f4d11ac016a
[
    {
        "Id": "2f4d11ac016a25718f46ae514218964bd0af70581154690ec679100e4be6a469",
        "Created": "2022-02-14T11:33:18.794029463Z",
        "Path": "python",
        "Args": [
            "./app.py"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 31930,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2022-02-15T13:01:34.457773995Z",
            "FinishedAt": "2022-02-15T10:55:25.924551668Z"
        },
        "Image": "sha256:c6c7f84ed0abe9cce6360d278ca895b5e1e35c7f76213dda655fa9e051bc38b2",
        "ResolvConfPath": "/opt/docker/containers/2f4d11ac016a25718f46ae514218964bd0af70581154690ec679100e4be6a469/resolv.conf",
        "HostnamePath": "/opt/docker/containers/2f4d11ac016a25718f46ae514218964bd0af70581154690ec679100e4be6a469/hostname",
        "HostsPath": "/opt/docker/containers/2f4d11ac016a25718f46ae514218964bd0af70581154690ec679100e4be6a469/hosts",
        "LogPath": "",
        "Name": "/flask-pyats-genie",
        "RestartCount": 0,
        "Driver": "overlay2",
        "MountLabel": "system_u:object_r:svirt_sandbox_file_t:s0:c106,c509",
        "ProcessLabel": "system_u:system_r:svirt_lxc_net_t:s0:c106,c509",
        "AppArmorProfile": "",
        "ExecIDs": [
            "98624c919104827ed528d15dad90813510e7b276e488f25e8a7a463ddb27032b"
        ],
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "journald",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": {
                "80/tcp": [
                    {
                        "HostIp": "",
                        "HostPort": "81"
                    }
                ]
            },
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "docker-runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": -1,
            "OomKillDisable": false,
            "PidsLimit": 0,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0
        },
        "GraphDriver": {
            "Name": "overlay2",
            "Data": {
                "LowerDir": "/opt/docker/overlay2/084de7e37703b0408f836af628655354d986cc92045f949f2944407ae4c24049-init/diff:/opt/docker/overlay2/0dd48da73bb093d1c9593cc292a84b6b6b3bd4d4cde9bc6cc45fc04fe85653ff/diff:/opt/docker/overlay2/a9131868a3229dd6cf7513b4f89132a93db2dcd88a7e2ca79c682cea71fa2812/diff:/opt/docker/overlay2/294128ce59754586f1e267270ce9021f39633417098c93a2ac0baa9f64321ea2/diff:/opt/docker/overlay2/3d973188649cbd9b5a67a32e21d14baee26db655c772bd4f98305c5363495f52/diff:/opt/docker/overlay2/c7fb46b712423ada8176000b40de0ad98b21c7a3745c3733b006fef7304fb64c/diff:/opt/docker/overlay2/4e91f625c8a4c109099f37cb2ca0240896b970a1a97dda7971bc48ca4840056b/diff:/opt/docker/overlay2/ee975b362a63508a01560b7f7f854119e0e7fd4e99e4a942fcc418f85a3fb7db/diff:/opt/docker/overlay2/90d75b6dbbeb099225290a80713003941e210600a2614adc4fac8fb0cdedf767/diff:/opt/docker/overlay2/006a0124472a596bd0c3dd65032e51a9660faa7741d9dd987f3028b0ba5ff4e1/diff:/opt/docker/overlay2/731dab000895545300a9cf49e7fc952fc68a2d2f39176179962950a725cd6e59/diff:/opt/docker/overlay2/21cff007a71462a7e274198872adee279f9fc9f7ecd1959ebce7f10c50cbde0f/diff:/opt/docker/overlay2/bdec485be6304f679c7da14326d3addd74c5172d633f091591afc43bd0ca6b36/diff:/opt/docker/overlay2/d9e7c542c3c7217ff32f4aa6fada5e97d5ca067753590d8eb6d7d0c592d98d22/diff",
                "MergedDir": "/opt/docker/overlay2/084de7e37703b0408f836af628655354d986cc92045f949f2944407ae4c24049/merged",
                "UpperDir": "/opt/docker/overlay2/084de7e37703b0408f836af628655354d986cc92045f949f2944407ae4c24049/diff",
                "WorkDir": "/opt/docker/overlay2/084de7e37703b0408f836af628655354d986cc92045f949f2944407ae4c24049/work"
            }
        },
        "Mounts": [],
        "Config": {
            "Hostname": "2f4d11ac016a",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "80/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "LANG=C.UTF-8",
                "GPG_KEY=0D96DF4D4110E5C43FBFB17F2D347EA6AA65421D",
                "PYTHON_VERSION=3.7.4",
                "PYTHON_PIP_VERSION=19.3",
                "PYTHON_GET_PIP_URL=https://github.com/pypa/get-pip/raw/65986a26949050d26e6ec98915da4aade8d8679d/get-pip.py",
                "PYTHON_GET_PIP_SHA256=8d412752ae26b46a39a201ec618ef9ef7656c5b2d8529cdcbe60cd70dc94f40c"
            ],
            "Cmd": [
                "python",
                "./app.py"
            ],
            "ArgsEscaped": true,
            "Image": "flask-image",
            "Volumes": null,
            "WorkingDir": "/code",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {}
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "cf29587dbc30fe0299acc2502a37edfab131b5017c514527a9c9fb232052d3a6",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "80/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "81"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/cf29587dbc30",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "043212671e8f97dc5f6043ffe8f504c2e38a25192438e5147d6d8fb90ffe763f",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.2",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:02",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "0a2e406f86cc73c27d25ef8072f50ef6f2e45b107d9b7ec8945e3c99c3ea7b1e",
                    "EndpointID": "043212671e8f97dc5f6043ffe8f504c2e38a25192438e5147d6d8fb90ffe763f",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.2",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:02"
                }
            }
        }
    }
]

**Reply by bwateratmsft** [reply]
Thanks. It looks like the `Platform` field is missing from the inspect result, which is expected on this version of Docker. You should be able to upgrade it by following [Docker's instructions](https://docs.docker.com/engine/install/#server). In the meantime, most features should still work, just not container files browsing.
